packages/hardhat-foundry/src/index.ts

```diff
@@ -0,0 +1,179 @@
+  // Check foundry.toml presence. Don't warn when running foundry initialization task
+  if (!existsSync(path.join(config.paths.root, "foundry.toml"))) {
+    if (!process.argv.includes(TASK_INIT_FOUNDRY)) {
+      console.log(
+        chalk.yellow(
+          `Warning: No foundry.toml file found at ${config.paths.root}, hardhat-foundry plugin will not activate. Consider running 'npx hardhat ${TASK_INIT_FOUNDRY}'`
+        )
+      );
+    }
+    return;
+  }
+
+  // Load foundry config
+  const foundryConfig = JSON.parse(runCmdSync("forge config --json"));
+
+  // Ensure required keys exist
+  if (
+    foundryConfig?.src === undefined ||
+    foundryConfig?.cache_path === undefined
+  ) {
+    throw new HardhatFoundryError(
+      "Couldn't find `src` or `cache_path` config keys after running `forge config --json`"
+    );
+  }
+
+  // Ensure foundry src path doesnt mismatch user-configured path
+  const userSourcesPath = userConfig.paths?.sources;
+  const foundrySourcesPath = foundryConfig.src;
+
+  if (userSourcesPath !== undefined && userSourcesPath !== foundrySourcesPath) {
+    throw new HardhatFoundryError(
+      `User-configured sources path (${userSourcesPath}) doesn't match path configured in foundry (${foundrySourcesPath})`
+    );
+  }
+
+  // Set sources path
+  config.paths.sources = foundryConfig.src;
+
+  // Change hardhat's cache path if it clashes with foundry's
+  if (foundryConfig.cache_path === "cache") {
+    config.paths.cache = "cache_hardhat";
+  }
+
+  pluginActivated = true;
+});
+
+// Task that translates import names to sourcenames using remappings
+internalTask(TASK_COMPILE_TRANSLATE_IMPORT_NAME).setAction(
+  async (
+    { importName }: { importName: string },
+    _hre,
+    runSuper
+  ): Promise<string> => {
+    if (!pluginActivated) {
+      return runSuper({ importName });
+    }
+
+    const remappings = getRemappings();
+
+    for (const from in remappings) {
+      if (Object.prototype.hasOwnProperty.call(remappings, from)) {
+        const to = remappings[from];
+        if (importName.startsWith(from)) {
+          return importName.replace(from, to);
+        }
+      }
+    }
+
+    return importName;
+  }
+);
+
+// Task that includes the remappings in solc input
+internalTask(TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE).setAction(
+  async (
+    {
+      dependencyGraph,
+      file,
+    }: {
+      dependencyGraph: DependencyGraph;
+      file: ResolvedFile;
+      solidityFilesCache?: SolidityFilesCache;
+    },
+    hre,
+    runSuper
+  ): Promise<CompilationJob | CompilationJobCreationError> => {
+    const job = (await runSuper({ dependencyGraph, file })) as
+      | CompilationJob
+      | CompilationJobCreationError;
+
+    if (!pluginActivated || isCompilationJobCreationError(job)) {
+      return job;
+    }
+
+    const remappings = getRemappings();
+    job.getSolcConfig().settings.remappings = Object.entries(remappings).map(
+      (fromTo) => fromTo.join("=")
+    );
```

Comment:
Personal preference:

```suggestion
    job.getSolcConfig().settings.remappings = Object.entries(remappings).map(
      ([from, to]) => `${from}=${to}`
    );
```

Two reasons: first, I think destructuring the key/value of `Object.entries` is common practice. Second, `.join` suggests to me an array of arbitrary length, so the meaning of the code is slightly less obvious.